File: next-dent-client/src/types/presupuesto.ts
export interface PresupuestoDetalle {
  idDetalle?: number;
  numeroFdi: number;
  idTarifa: number;
  precioUnitario: number;
  estado?: string;
  motivoAnulacion?: string;
}

export interface Presupuesto {
  idPresupuesto?: number;
  idPaciente: number;
  fechaCreacion: string;
  total: number;
  estado: string;
  detalles: PresupuestoDetalle[];
}

export interface PresupuestoDetalleRequest {
  numeroFdi: number;
  idTarifa: number;
  precioUnitario: number;
}

export interface PresupuestoCreateRequest {
  idPaciente: number;
  detalles: PresupuestoDetalleRequest[];
}

File: next-dent-api/next-dent-api/src/main/java/com/clinica/next_dent_api/model/Presupuesto.java
package com.clinica.next_dent_api.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tb_presupuesto")
public class Presupuesto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_presupuesto")
    private Integer idPresupuesto;

    @Column(name = "id_paciente", nullable = false)
    private Integer idPaciente;

    @Column(name = "fecha_creacion", nullable = false)
    private LocalDateTime fechaCreacion;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal total;

    @Column(nullable = false, length = 20)
    private String estado;

    @JsonManagedReference
    @OneToMany(mappedBy = "presupuesto", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PresupuestoDetalle> detalles;
}
